nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage.java

```diff
@@ -141,14 +142,14 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 }
 
                 try {
-                    blob.upload(in, -1, null, null, operationContext);
+                    uploadBlob(blob, operationContext, in);
                     BlobProperties properties = blob.getProperties();
                     attributes.put("azure.container", containerName);
                     attributes.put("azure.primaryUri", blob.getSnapshotQualifiedUri().toString());
                     attributes.put("azure.etag", properties.getEtag());
                     attributes.put("azure.length", String.valueOf(length));
                     attributes.put("azure.timestamp", String.valueOf(properties.getLastModified()));
-                } catch (StorageException | URISyntaxException e) {
+                } catch (StorageException | URISyntaxException | IOException e) {
                     storedException.set(e);
```

Comment:
Reviewing the larger context of this block, is there a reason to continue maintaining this reference to the `storageException`?  The checked `StorageException` and `URISyntaxException` could be wrapped in the existing `throw new IOException()`.  If there is a particular reason to evaluate or log the `StorageException`, that could be handled here, or in the outer catch block.  Either way, it seems like it would be worth untangling the exception handling to avoid this kind of scenario down the road.

Reply:
@exceptionfactory, thanks for the comment. I have checked the bigger context, and although I agree that the `storedException` is not needed here, but when the original exception gets propagated to the [outer `catch` block](https://github.com/apache/nifi/blob/c8b7bf9b427b1a2026d03c4d3438a5238da8acd2/nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage.java#L167) it's wrapped into a `ProcessException` by the framework.
This means that to get the original exception something like `if (e instanceof ProcessException && e.getCause() instanceof IOException && e.getCause().getCause() != null)...` would be needed.
In my opinion the current logic is cleaner and easier to understand.

Reply:
Thanks for the response @adenes.  I was actually wondering whether that check for `ProcessException` in the outer catch block should be there at all.  As it stands, anything that throws a `ProcessException` will keep FlowFiles in the queue instead of routing to failure.  Perhaps that is the desired behavior with this particular Processor, but if it is not necessary, the exception handling could be generalized to avoid the `instanceof` check so that all exceptions result in routing to failure.  I am fine with the changes as implemented now, but wanted to raise this question given the current issue with exception handling.

Reply:
I looked at the blame to see what the original intent is and it's not 100% clear. I have to assume then that `REL_FAILURE` is intended to be specifically for Blob Storage or Blob Storage operation failures and nothing else.

I think we should leave as-is for now. We really need to get back to #4430 or similar to get this on more current dependencies. We can probably (re)think through the "right" exception handling there.

Reply:
Thanks for the background @jfrazee, leaving it as-is for now sounds good.